src/main/java/com/google/sps/Utility.java

```diff
@@ -207,6 +205,35 @@ public static MultiMutation getMultiMutationAtIndex(List<MultiMutation> multiMut
     return lst;
   }
 
+  /**
+   * Returns a set of indices on the original list that related to a given token
+   *
+   * @param tokenName the token name to search for
+   * @param origList the original list of mutations
+   * @return a set of indices, empty if tokenName is null or if token is not changed
+   */
+  public static Set<Integer> getMutationIndicesOfToken(
```

Comment:
This also seems like a potential optimization opportunity: store a mapping from token to mutation indices?